tests/unit/ml_handlers/test_rag.py

```diff
@@ -0,0 +1,196 @@
+        # create project
+
+        self.run_sql("create database proj")
+        self.run_sql(
+            f"""
+              create model proj.test_rag_openai_nonexistant_model
+              predict answer
+              using
+                engine='rag',
+                llm_type='openai',
+                model_id='this-model-does-not-exist',
+                openai_api_key='{OPENAI_API_KEY}';
+           """
+        )
+        with pytest.raises(Exception):
+            self.wait_predictor("proj", "test_rag_openai_nonexistant_model")
+
+        self.run_sql(
+            f"""
+                  create model proj.test_rag_writer_nonexistant_model
+                  predict answer
+                  using
+                    engine='rag',
+                    llm_type='writer',
+                    model_id='this-model-does-not-exist',
+                    writer_api_key='{WRITER_API_KEY}',
+                    writer_org_id='{WRITER_ORG_ID}';
+               """
+        )
+
+        with pytest.raises(Exception):
+            self.wait_predictor("proj", "test_rag_writer_nonexistant_model")
+
+    def test_unsupported_llm_type(self):
+        self.run_sql("create database proj")
+        self.run_sql(
+            """
+            create model proj.test_unsupported_llm
+            predict answer
+            using
+                engine='rag',
+                llm_type='unsupported_llm'
+        """
+        )
+        with pytest.raises(Exception):
+            self.wait_predictor("proj", "test_unsupported_llm")
+
+    def test_unsupported_vector_store(self):
+        self.run_sql("create database proj")
+        self.run_sql(
+            f"""
+            create model proj.test_unsupported_vector_store
+            predict answer
+            using
+                engine='rag',
+                llm_type='openai',
+                openai_api_key='{OPENAI_API_KEY}',
+                vector_store_name='unsupported_vector_store'
+        """
+        )
+
+        with pytest.raises(Exception):
+            self.wait_predictor("proj", "test_unsupported_vector_store")
+
+    def test_unknown_arguments(self):
+        self.run_sql("create database proj")
+        self.run_sql(
+            f"""
+            create model proj.test_openai_unknown_arguments
+            predict answer
+            using
+                engine='rag',
+                llm_type='openai',
+                openai_api_key='{OPENAI_API_KEY}',
+                evidently_wrong_argument='wrong value'  --- this is a wrong argument name
+        """
+        )
+        with pytest.raises(Exception):
+            self.wait_predictor("proj", "test_openai_unknown_arguments")
+
+    @pytest.mark.xfail(
+        reason="there seems to be an issue with running inner queries, it appears to be a potential bug in the mock handler"
+    )
+    @patch("mindsdb.integrations.handlers.postgres_handler.Handler")
+    def test_qa(self, postgres_mock_handler):
```

Comment:
@paxcema seeing weird behaviour on this test, also across other tests. Unclear whether its an issue with my setup or a wider issue. Another strong argument for  handler unit on push CI/CD at the very least

Reply:
Awesome, agreed. Let's go ahead with this PR and solve that issue soon.